app/main.py

```diff
@@ -2,24 +2,69 @@
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: type, name: str) -> None:
+        self._protected_name = "_" + name
+
+    def __get__(self, instance: object, owner: type) -> int:
+        return getattr(instance, self._protected_name)
+
+    def __set__(self, instance: object, value: int) -> None:
+        if (self.min_amount <= value <= self.max_amount):
+            setattr(instance, self._protected_name, value)
+        else:
```

Comment:
You don't need this `else` here, better to raise an error it not (self.min_amount <= value <= self.max_amount)